hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3secret/S3SecretGenerateEndpoint.java

```diff
@@ -34,16 +36,30 @@
 public class S3SecretGenerateEndpoint extends S3SecretEndpointBase {
   @POST
   public Response generate() throws IOException {
+    return generateInternal(null);
+  }
+
+  @POST
+  @Path("/{username}")
+  public Response generate(@PathParam("username") String username)
```

Comment:
The current URL and HTTP action of each operation is 
POST /secret/generate
POST /secret/generate/testuser2
POST /secret/revoke
POST /secret/revoke/testuser2

For a REST API, it usually leverages HTTP operations instead of put the operation/action name in the URL. I'd like to recommend change the URL and HTTP action to,
PUT        /secret/
PUT        /secret/testuser2
DELETE  /secret/
DELETE  /secret/testuser2

Reply:
Yes, it's good to have a page for this new RESTful solution. 

There is one Security section on this page https://ci-hadoop.apache.org/view/Hadoop%20Ozone/job/ozone-doc-master/lastSuccessfulBuild/artifact/hadoop-hdds/docs/public/interface/s3.html.  about the CLI S3 Secret Management. It's better to co locate the new content with it.  Maybe we can move the Security section content from s3.html to the Securing S3.html, then add a link in s3.html to refer to the new location.